src/HeaderAddPlugin.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Adds headers to the request only if the request does not already own it.
```

Comment:
Well, afterthough I think we should rename ``HeaderAddPlugin`` to ``HeaderAppendPlugin``.

We can also implement ``HeaderAddPlugin``, but I think it's an edge case, and maybe custom plugin should be used for that?

I'm based on the example of the user agent. By default I want an user agent string, but if the request already has an user agent string, I wont put 2 values for user agent, I will set the header only if the request doesn't already has one.